### PR TITLE
DT-2429: 💄 Switch to sign in / out instead of login / out

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: backend/sessionManagementRoutes.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ const configureRoutes = ({ app, tokenRefresher, tokenVerifier, homeLink }) => {
       }
       next()
     } catch (error) {
-      // need to logout here otherwise user will still be considered authenticated when we take them to /login
+      // need to sign out here otherwise user will still be considered authenticated when we take them to /sign-in
       req.logout()
```
Comment (petergphillips) [inline backend/sessionManagementRoutes.js:61]: function on `req` is provided by passport, so still need to remain as `logout`

File: backend/sessionManagementRoutes.js [hunk 2]
```diff
@@ -95,13 +95,17 @@ const configureRoutes = ({ app, tokenRefresher, tokenVerifier, homeLink }) => {
...
+      res.redirect(`/sign-in?${query}`)
     })
   }
 
-  app.get('/login', loginMiddleware, remoteLoginIndex)
+  app.get('/sign-in', signInMiddleware, remoteSignInIndex)
 
-  app.get('/login/callback', (req, res, next) => {
+  app.get('/login/callback', (req, res) => {
+    res.redirect('/sign-in/callback')
```
Comment (petergphillips) [inline backend/sessionManagementRoutes.js:105]: add in redirect if user has bookmarked old url in auth

File: backend/sessionManagementRoutes.js [hunk 3]
```diff
@@ -115,7 +119,7 @@ const configureRoutes = ({ app, tokenRefresher, tokenVerifier, homeLink }) => {
         logger.info(`Auth failure due to ${JSON.stringify(info)}`)
         return res.redirect('/autherror')
       }
-      req.logIn(user, (err2) => {
+      req.login(user, (err2) => {
```
Comment (petergphillips) [inline backend/sessionManagementRoutes.js:122]: these are override of passport function so no changed to sign in.  Can be called with either `logIn` or `login` - see https://github.com/jaredhanson/passport/blob/master/lib/http/request.js#L31

File: integration-tests/integration/login.spec.js [hunk 4]
```diff
@@ -9,39 +9,49 @@ context('Login functionality', () => {
...
     MenuPage.verifyOnPage()
 
     // can't do a visit here as cypress requires only one domain
-    cy.request('/auth/logout').its('body').should('contain', 'Sign in')
+    cy.request('/auth/sign-out').its('body').should('contain', 'Sign in')
+  })
+
+  it('Direct access to login callback takes user to sign in page', () => {
+    cy.task('stubSignIn', {})
+    cy.visit('/login/callback')
```
Comment (petergphillips) [inline integration-tests/integration/login.spec.js:44]: test of new redirect

File: docker-compose.yaml [hunk 5]
```diff
@@ -11,6 +11,7 @@ services:
       test: ['CMD', 'curl', '-f', 'http://localhost:8080/auth/health']
     environment:
       - SPRING_PROFILES_ACTIVE=dev
+      - APPLICATION_AUTHENTICATION_UI_ALLOWLIST=0.0.0.0/0
```
Comment (petergphillips) [inline docker-compose.yaml:14]: this allows access to `/auth/ui` so that client configuration can be modified / viewed

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**